intro-to-debugging/intro-to-debugging.md

```diff
@@ -0,0 +1,207 @@
+
+Just be sure to take note of the syntax: inside of the parentheses in `print()`, we should separate each value with a comma.
+
+For example, observe this code:
+
+```python
+variable_message = "variables"
+print("We", "can", "give", "this", "many", "strings", "or", variable_message, "or", "expressions:", 4+3, len("rainbow"))
+```
+
+More practically, we might use it like this:
+
+```python
+apples = "red"
+oranges = "orange"
+blueberries = "blue"
+
+print("apples is:", apples, "oranges is:", oranges, "blueberries is:", blueberries)
+```
+
+### Rubber Ducking
+
+[Rubber Duck Debugging, or Rubber Ducking](https://en.wikipedia.org/wiki/Rubber_duck_debugging) is a method of debugging code. It describes the idea that sometimes the best way to debug code is to talk things through out loud, and that sometimes talking out loud to a rubber duck will do.
+
+Sometimes, the solution to your problem is solved just by trying to explain your code out loud.
+
+The idea of Rubber Ducking is common and often joked about, to the point that many software development teams will provide literal rubber ducks to developers.
+
+## All Together: Ask and Answer Questions with Debugging
+
+When we combine all these concepts, we get:
+
+- a generic flowchart of how to debug, and what questions to ask and answer
+- a generic set of some debugging tools to help us ask and answer those questions
+
+We can even break it up this way:
+
+1. Figure out **what** is happening
+    - Use your error messages and `print` statements
+2. Figure out **why** it's happening
+    - Use `print` statements, rubber ducking skills, and other future debuging tools
+3. Figure out **how** to fix it
+    - Make theories and then test them!
+    - Use rubber ducking skills, experimentation, research, and ask for help!
+
+In conclusion, debugging is one of the main things we'll be doing as programmers. We will learn so many more debugging tools in the future!
+
+## Check for Understanding
+
+Select all methods that are valid ways to approach debugging
+
+
+* Asking "what" is happening
+* Talking to a literal rubber duck
+* Talking to a classmate
+* Writing down what you expected to happen vs. what actually happened
+* Taking a break and looking at it again in an hour
+* Using `print` statements to confirm the value of a variable on a given line
+* Making theories and experiments on why the bug is there and how to fix it
+* Taking notes about the bug
+* Making a table or diagram
+
+
+* Asking "what" is happening
+* Talking to a literal rubber duck
+* Talking to a classmate
+* Writing down what you expected to happen vs. what actually happened
+* Taking a break and looking at it again in an hour
+* Using `print` statements to confirm the value of a variable on a given line
+* Making theories and experiments on why the bug is there and how to fix it
+* Taking notes about the bug
+* Making a table or diagram
```

Comment:
```suggestion
### !challenge
* type: checkbox
* id: 27fc1277-cb27-426a-aff8-0298638fb5f0
* title: Debugging Methods

##### !question
Select all methods that are valid ways to approach debugging.
##### !end-question

##### !options
* Asking "what" is happening
* Talking to a literal rubber duck
* Talking to a classmate
* Writing down what you expected to happen vs. what actually happened
* Taking a break and looking at it again in an hour
* Using `print` statements to confirm the value of a variable on a given line
* Making theories and experiments on why the bug is there and how to fix it
* Taking notes about the bug
* Making a table or diagram
##### !end-options

##### !answer
* 
##### !end-answer

##### !explanation 
These are all great debugging methods!
##### !end-explanation 

### !end-challenge
```

Reply:
I thought all of these were valid, let me know if I'm wrong lol!

Reply:
They are!!! :D Thanks